### PR TITLE
Simplify icon related alignments and paddings

### PR DESCRIPTION
**Pull request by PavelHolec**
#317 

Text line height is updated to match designs (independently on specific font). This makes it possible to simplify icon alignment, as now the `.top` alignment can be used for same-size icon and text. 

The `firstBaseline` alignment for icons is still supported, but the calculation is not as precise, so it is not used in most components, unless needed (`BadgeList`).

image

Icon alignment is simplified (all vertical guidelines can be used) as the icon size is equal to line height

image

Component sizes match design sizing independently on used font

image

image

image

### CODE REVIEW

File: Sources/Orbit/Components/Tag.swift [hunk 1]
```diff
@@ -3,10 +3,11 @@ import SwiftUI
 /// Offers a label that can optionally be selected and unselected or removed.
 ///
 /// - Note: [Orbit definition](https://orbit.kiwi/components/tag/)
+/// - Important: Component can expand horizontally by using `idealSize` set to false.
```
Comment (sjavora) [inline Sources/Orbit/Components/Tag.swift:6]: Maybe
> Component will expand horizontally if `idealSize` is set to `false`

?

**Comment by sjavora** [comment]
Very nice!

Seeing stuff like
Screenshot 2022-12-31 at 22 09 25
...shouldn't `.measured()` be limited to a single line?

File: Sources/Orbit/Components/Tag.swift (hunk 1 above)
Comment (PavelHolec) [reply Sources/Orbit/Components/Tag.swift]: > Very nice!
> 
> Seeing stuff like Screenshot 2022-12-31 at 22 09 25 ...shouldn't `.measured()` be limited to a single line?

Thx, updated with a single line limit and previews.

File: Sources/Orbit/Components/Icon.swift [hunk 2]
```diff
@@ -17,48 +20,98 @@ public struct Icon: View {
...
+    }
+
+    @ViewBuilder func imageWrapper<Content: View>(mode: ContentMode, @ViewBuilder content: () -> Content) -> some View {
+        content()
+            .aspectRatio(contentMode: mode)
+            .frame(width: dynamicSize, height: dynamicSize)
+            .accessibility(hidden: true)
+    }
+
+    @ViewBuilder func sfSymbolWrapper<Content: View>(weight: Font.Weight, @ViewBuilder content: () -> Content) -> some View {
```
[the remark this answers is not in the document]
Comment (PavelHolec) [reply Sources/Orbit/Components/Icon.swift]: It is, 👍 Seems like this was already replaced in other files.

File: Sources/Orbit/Components/Icon.swift [hunk 3]
```diff
@@ -5,8 +5,11 @@ import SwiftUI
 /// - Note: [Orbit definition](https://orbit.kiwi/components/icon/)
 public struct Icon: View {
 
-    public static let sfSymbolToOrbitSizeRatio: CGFloat = 0.75
-    public static let averageIconContentPadding: CGFloat = .xxxSmall
+    public static let sfSymbolToOrbitSymbolSizeRatio: CGFloat = 0.75
```
[the remark this answers is not in the document]
Comment (PavelHolec) [reply Sources/Orbit/Components/Icon.swift:9]: Added and removed unused countryFlag baseline constant.

File: Sources/Orbit/Support/TextConcatenation/TextRepresentable+Concatenation.swift [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
-/// Conctatenates two terms that can be represented as `SwiftUI.Text`
+/// Concatenates two terms that can be represented as `SwiftUI.Text`
 ///
 /// - Parameters:
 ///   - left: A content representable as `SwiftUI.Text`
 ///   - right: A content representable as `SwiftUI.Text`
 /// - Returns: A view that is the result of concatenation of text representation of the parameters.
-///   if both paramters do not have a text representation, the returning view will produce `EmptyView`, preserving the standard Orbit behavior
+///   if both parameters do not have a text representation, the returning view will produce `EmptyView`, preserving the standard Orbit behavior
```
[the remark this answers is not in the document]
Comment (PavelHolec) [reply Sources/Orbit/Support/TextConcatenation/TextRepresentable+Concatenation.swift]: > Very nice!
> 
> Seeing stuff like Screenshot 2022-12-31 at 22 09 25 ...shouldn't `.measured()` be limited to a single line?

Thx, updated with a single line limit and previews.

**Comment by PavelHolec** [comment]
> Very nice!
> 
> Seeing stuff like Screenshot 2022-12-31 at 22 09 25 ...shouldn't `.measured()` be limited to a single line?

Thx, updated with a single line limit and previews.

**Review by sjavora — APPROVED** [review]

**Pull request merged**